src/tsdb.c

```diff
@@ -511,12 +493,19 @@ int MultiSerieReduce(Series *dest,
 }
 
 static bool RuleSeriesUpsertSample(RedisModuleCtx *ctx,
+                                   Series *series,
                                    CompactionRule *rule,
                                    timestamp_t start,
                                    double val) {
     RedisModuleKey *key;
     Series *destSeries;
-    if (!GetSeries(ctx, rule->destKey, &key, &destSeries, REDISMODULE_READ | REDISMODULE_WRITE)) {
+    if (!GetSeries(ctx,
+                   rule->destKey,
+                   &key,
+                   &destSeries,
+                   REDISMODULE_READ | REDISMODULE_WRITE,
+                   false,
+                   false)) {
```

Comment:
If we upsert and change history which will be written to the compaction rules don't we want to clean the rules before?

Reply:
Oh you moved it to `upsertCompaction`, ok then.